rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```
@@ -441,30 +461,111 @@ BtcBlockStoreWithCache getBtcBlockStore() {
+            return new PeginEvaluationResult(PeginProcessAction.CANNOT_BE_PROCESSED, NO_UTXO_OR_UTXO_BELOW_MINIMUM);
+        }
 
-        Wallet federationWallet = new BridgeBtcWallet(btcContext, getLiveFederations());
+        try {
+            peginInformation.parse(btcTx);
+        } catch (PeginInstructionsException e) {
+            String message = String.format(
+                "Error while trying to parse peg-in information for tx %s. %s",
+                btcTx.getHash(),
+                e.getMessage()
+            );
+            logger.warn("[evaluatePegin] {}", message);
+            if (peginInformation.getBtcRefundAddress() != null){
+                return new PeginEvaluationResult(PeginProcessAction.CAN_BE_REFUNDED, PEGIN_V1_INVALID_PAYLOAD);
+            } else {
+                return new PeginEvaluationResult(PeginProcessAction.CANNOT_BE_PROCESSED, PEGIN_V1_INVALID_PAYLOAD);
+            }
+        }
 
-        return PegUtilsLegacy.getTransactionType(
-            btcTx,
-            getActiveFederation(),
-            getRetiringFederation(),
-            retiredFederationP2SHScript,
-            oldFederationAddress,
-            activations,
-            minimumPeginTxValue,
-            federationWallet
+        if (peginInformation.getProtocolVersion() == 0) {
+            switch (peginInformation.getSenderBtcAddressType()) {
+                case P2PKH:
+                case P2SHP2WPKH:
+                    return new PeginEvaluationResult(PeginProcessAction.CAN_BE_REGISTERED);
+                case P2SHMULTISIG:
+                case P2SHP2WSH:
+                    return new PeginEvaluationResult(PeginProcessAction.CAN_BE_REFUNDED, LEGACY_PEGIN_MULTISIG_SENDER);
+                default:
+                    return new PeginEvaluationResult(PeginProcessAction.CANNOT_BE_PROCESSED, LEGACY_PEGIN_UNDETERMINED_SENDER);
```

If the sender could not be determined then the pegin information parser would have thrown an exception, right?

In that case we are not expecting to reach this point. I guess we should throw an exception if this happens